programs/stake/src/stake_instruction.rs

```diff
@@ -36,22 +36,49 @@ pub fn process_instruction(
     trace!("process_instruction: {:?}", data);
     trace!("keyed_accounts: {:?}", keyed_accounts);
 
-    let me = &keyed_account_at_index(keyed_accounts, first_instruction_account)?;
-    if me.owner()? != id() {
-        return Err(InstructionError::InvalidAccountOwner);
-    }
+    let (me, stake_instruction) = if invoke_context
+        .feature_set
+        .is_active(&feature_set::add_get_minimum_delegation_instruction_to_stake_program::id())
+    {
+        let stake_instruction = limited_deserialize(data)?;
+        let me = if matches!(stake_instruction, StakeInstruction::GetMinimumDelegation) {
+            None
+        } else {
+            let me = keyed_account_at_index(keyed_accounts, first_instruction_account)?;
+            if me.owner()? != id() {
+                return Err(InstructionError::InvalidAccountOwner);
+            }
+            Some(me)
+        };
+        (me, stake_instruction)
+    } else {
+        let me = keyed_account_at_index(keyed_accounts, first_instruction_account)?;
+        if me.owner()? != id() {
+            return Err(InstructionError::InvalidAccountOwner);
+        }
+        let stake_instruction = limited_deserialize(data)?;
+        (Some(me), stake_instruction)
+    };
```

Comment:
Here's the new code. Since the *order* of errors needs to stay the same if the `add_get_minimum_delegation_instruction_to_stake_program` feature is not enabled, I had to do some ugliness to retain behavior. Definitely open to alternative implementations here.

Reply:
Here's an alternative implementation: https://github.com/brooksprumo/solana/pull/1

Reply:
Thanks! I like this much more; merged.